### PR TITLE
[core] Deprecate utility classes

### PR DESCRIPTION
**Pull request by oowekyala**
Deprecate contents of nspmd.util as they're internal API. Many of these can be completely removed with 7.0.0, eg Filters, EmptyIterator, FileIterable, etc. Sometimes there's already an alternative in JDK 7.

### CODE REVIEW

File: pmd-core/src/main/java/net/sourceforge/pmd/util/FileUtil.java [hunk 1]
```diff
@@ -148,21 +154,27 @@ public static boolean findPatternInFile(final File file, final String pattern) {
         Pattern regexp = Pattern.compile(pattern);
         Matcher matcher = regexp.matcher("");
 
-        FileIterable it = new FileIterable(file);
-        for (String line : it) {
-            matcher.reset(line); // reset the input
-            if (matcher.find()) {
-                return true;
+        try {
+            for (String line : Files.readAllLines(file.toPath())) {
```
Comment (jsotuyod) [inline pmd-core/src/main/java/net/sourceforge/pmd/util/FileUtil.java]: this call requires a charset in Java 7. The single arg version was added in Java 8 and is unavailable in master

File: pmd-core/src/main/java/net/sourceforge/pmd/Report.java [hunk 2]
```diff
@@ -491,7 +490,7 @@ public boolean treeIsEmpty() {
      * @return the iterator
      */
     public Iterator<ProcessingError> errors() {
-        return errors == null ? EmptyIterator.<ProcessingError>instance() : errors.iterator();
+        return errors == null ? Collections.emptyIterator() : errors.iterator();
```
Comment (jsotuyod) [inline pmd-core/src/main/java/net/sourceforge/pmd/Report.java]: this should be 
```suggestion
        return errors == null ? Collections<ProcessingError>.emptyIterator() : errors.iterator();
```

File: pmd-core/src/main/java/net/sourceforge/pmd/Report.java [hunk 3]
```diff
@@ -500,7 +499,7 @@ public boolean treeIsEmpty() {
      * @return the iterator
      */
     public Iterator<ConfigurationError> configErrors() {
-        return configErrors == null ? EmptyIterator.<ConfigurationError>instance() : configErrors.iterator();
+        return configErrors == null ? Collections.emptyIterator() : configErrors.iterator();
```
Comment (jsotuyod) [inline pmd-core/src/main/java/net/sourceforge/pmd/Report.java]: this should be
```suggestion
        return configErrors == null ? Collections<ConfigurationError>.emptyIterator() : configErrors.iterator();
```

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">This changeset introduces 0 new violations, 0 new errors and 0 new configuration errors,<br />
removes 0 violations, 0 errors and 0 configuration errors.<br />
<a href="https://chunk.io/pmd/67cddb9b4e5d4ea1b760b5c7f2d9dd6b/diff/index.html">Full report</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: pmd-core/src/main/java/net/sourceforge/pmd/cpd/GridBagHelper.java [hunk 4]
```diff
@@ -13,6 +13,13 @@
 import javax.swing.JLabel;
 import javax.swing.SwingConstants;
 
+import net.sourceforge.pmd.annotation.InternalApi;
+
+/**
+ * @deprecated Is internal API
+ */
+@Deprecated
+@InternalApi
```
Comment (adangel) [inline pmd-core/src/main/java/net/sourceforge/pmd/cpd/GridBagHelper.java]: This should be added in the release notes, just for completeness

File: pmd-core/src/main/java/net/sourceforge/pmd/renderers/ColumnDescriptor.java [hunk 5]
```diff
@@ -4,12 +4,17 @@
...
+import net.sourceforge.pmd.annotation.InternalApi;
+
 /**
  *
  * @author Brian Remedios
  *
  * @param <T>
+ * @deprecated Is internal API
  */
+@InternalApi
```
Comment (adangel) [inline pmd-core/src/main/java/net/sourceforge/pmd/renderers/ColumnDescriptor.java]: Ditto

File: pmd-core/src/main/java/net/sourceforge/pmd/util/ClassUtil.java [hunk 6]
```diff
@@ -17,7 +17,9 @@
  * UIs without the package prefixes.
  *
  * @author Brian Remedios
+ * @deprecated Is internal API
  */
+@Deprecated
```
Comment (adangel) [inline pmd-core/src/main/java/net/sourceforge/pmd/util/ClassUtil.java]: Why not `@InternalApi`?

Comment (oowekyala) [reply pmd-core/src/main/java/net/sourceforge/pmd/util/ClassUtil.java]: It looks like this was used only by type and method properties, which are deprecated for removal.

**Pull request merged**